pinot-core/src/main/java/com/linkedin/pinot/core/query/scheduler/QueryScheduler.java

```diff
@@ -135,39 +136,38 @@ public void stop() {
     byte[] responseData = serializeDataTable(queryRequest, dataTable);
 
     // Log the statistics
-    long numDocsScanned = 0;
-    long numEntriesScannedInFilter = 0;
-    long numEntriesScannedPostFilter = 0;
     String tableNameWithType = queryRequest.getTableNameWithType();
-    try {
-      numDocsScanned = Long.parseLong(getMetadataValue(dataTableMetadata, DataTable.NUM_DOCS_SCANNED_METADATA_KEY));
-      numEntriesScannedInFilter = Long.parseLong(getMetadataValue(dataTableMetadata, DataTable.NUM_ENTRIES_SCANNED_IN_FILTER_METADATA_KEY));
-      numEntriesScannedPostFilter = Long.parseLong(getMetadataValue(dataTableMetadata, DataTable.NUM_ENTRIES_SCANNED_POST_FILTER_METADATA_KEY));
+    long numDocsScanned =
+        Long.parseLong(dataTableMetadata.getOrDefault(DataTable.NUM_DOCS_SCANNED_METADATA_KEY, INVALID_NUM_SCANNED));
+    long numEntriesScannedInFilter = Long.parseLong(
+        dataTableMetadata.getOrDefault(DataTable.NUM_ENTRIES_SCANNED_IN_FILTER_METADATA_KEY, INVALID_NUM_SCANNED));
```

Comment:
We are no longer catching parse exception. This seems fine for the most part (ie, it shouldn't really happen), but if it does then we fail the query while trying to do something non-essential. Something to consider - do we need to be more defensive?

Reply:
This should be okay, unless we explicitly put non-number into this field, which is unlikely to happen.